igor-web/src/main/groovy/com/netflix/spinnaker/igor/jenkins/JenkinsBuildMonitor.groovy

```diff
@@ -190,14 +191,22 @@ class JenkinsBuildMonitor implements PollingMonitor {
 
                     if (!igorConfigurationProperties.spinnaker.build.processBuildsOlderThanLookBackWindow) {
                         use (TimeCategory) {
-                            completedBuilds = completedBuilds.findAll {
-                                def offsetMins = pollInterval.minutes
-                                def lookBackWindowMins = igorConfigurationProperties.spinnaker.build.lookBackWindowMins.minutes
-                                Date lookBackDate = (offsetMins + lookBackWindowMins).ago
-                                Date buildDate = new Date(it.timestamp as Long)
-                                log.info("filtering builds older than {}", lookBackDate)
-                                return buildDate.after(lookBackDate)
+                            def offsetSeconds = pollInterval.seconds
+                            def lookBackWindowMins = igorConfigurationProperties.spinnaker.build.lookBackWindowMins.minutes
+                            Date lookBackDate = (offsetSeconds + lookBackWindowMins).ago
+
+                            def tooOldBuilds = completedBuilds.findAll {
+                                Date buildEndDate = new Date(it.timestamp as Long)
+                                DateUtils.addMilliseconds(buildEndDate,Math.toIntExact(it.duration))
```

Comment:
I've never used `DateUtils.addMilliseconds()` but the javadoc indicates it returns a `Date`, we don't appaer to be using it?

@emjburns ^^

Reply:
ya good feedback. I"ll change to that. it.duration is the number of ms the build took)